documentation/projects/proposals/relicensing/20241028-implementation_plan_relicensing.md

```diff
@@ -0,0 +1,301 @@
+before the license change under the terms of the MIT license.
+
+By doing so, we are not betraying the trust of the contributors who have
+contributed to the project as their contributions will continue to be
+open-source. We did not have a CLA and will not add one.
+
+> Thus, the absence of a CLA combined with the use of a copyleft license serves
+> as a strong promise about the future of the project.
+>
+> - [Drew deVault's blog](https://drewdevault.com/2023/07/04/Dont-sign-a-CLA-2.html)
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+The outcomes of the implementation plan will be the names of the final license
+for each code block and the process we will use to migrate the license.
+
+## Step-by-step plan
+
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
+
+The goal of this section is to give a high-level view of the order of implementation any relationships like
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+1. Determine code blocks.
+2. Determine the goals for these blocks from the license.
+3. Determine new licenses.
+4. Institute code freeze.
+5. Apply new licenses.
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+### Determine code blocks
+
+The Openverse monorepo consists of the following code blocks.
+
+- Libraries
+  - Attribution
+  - API client
+- Applications
+  - API
+  - Frontend
+- DAGs
+- Glue
+  - Workflows
+  - Automations
+  - `ov` (development environment)
+
+### Determine the goals for these blocks from the license
+
+- Libraries
+
+  We want our libraries to be used as broadly as possible, in both free as well
+  as non-free software.
+
+- Applications
+
+  We want our applications to be open-source, and forks of these applications to
+  be open-source as well. We want these protections to apply even when these
+  applications are accessed over the Internet.
+
+### Determine new licenses
+
+- Libraries
+
+  The GNU LGPL license allows our libs to be used universally in both free as
+  well as non-free environments.
+
+- Applications
+
+  The GNU AGPL license requires derivatives of the API and frontend, which are
+  accessed over a network to also share source code, which is want we want.
+
+- Everything else
+
+  The GNU GPL license is a good fit for the rest of the codebase, since there
+  are no special requirements for this code (unlike apps and libs).
```

Comment:
Can we get an excerpt or link to what the terms of these licenses imply? It could be mentioned in the project proposal and linked here, but regardless of the place, it would be good to know where they're coming from.

Also, I wonder if the Django project and Airflow from Astronomer are compatible with the licenses. Can we get a clarification here?

Reply:
Added excerpts in 4d3e843. Airflow and Django are compatible with the new (A/L)?GPL licenses as they both use very permissible licenses (Apache and BSD-3-clause) respectively.